### DISCUSSION TITLE
How do I set my quarto website such that blog posts that I wrote ages ago are not re-rendered every time I build my site?

### DISCUSSION THREAD

**Discussion by Shelmith-Kariuki** [quarto-dev/quarto-cli#11323] [Q&A] [answered]
### Description

Hi, 

I have included the option ```freeze:auto``` on all my qmd files but every time I edit any file and render the site, all the files are rendered which is a slow and long process. Right now,  I'm only working on file 22 (see image attached) and would want only that file to be rendered.

![quarto-renders-all-files](https://github.com/user-attachments/assets/b1d7c1b5-f1d2-40ce-9be2-3e941f50393c)

**Comment by cscheid** [comment, marked as answer]
`freeze` should still be happening here. If you have a reproducible example where `freeze` is still causing `knitr` or `jupyter` to run, that's a Quarto bug that we need to work on.

The rendering you're seeing happens after the engine, and that goes independently of `freeze`. (We're aware that rendering in Quarto is slower than we'd like it to be, and we're working on it.)

**Comment by Shelmith-Kariuki** [comment]
Hi @cscheid 

Thank you for your response. 

Actually, none of my files contain any code. It's just a bunch of texts and images. 

Does it mean that the freeze mechanism works when there are code chunks in the qmd file?

**Reply by cscheid** [reply]
Yes. We should document this more clearly, but https://quarto.org/docs/projects/code-execution.html#freeze stores the results of execution engines, and not the final (eg) .html or PDF output.

**Reply by Shelmith-Kariuki** [reply]
Aaah, okay. Now I understand. Thank you.
